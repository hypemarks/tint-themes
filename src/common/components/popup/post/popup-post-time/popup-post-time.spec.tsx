import { render } from '@testing-library/react';
import React from 'react';
import { buildPost } from '../../../../../../test/generate';
import PopupPostTime from './popup-post-time';

describe('Test <PopupPostImage /> Component', () => {
  const post = buildPost({ attributes: { published_at: null } });

  it('should render snapshot', () => {
    const wrapper = render(<PopupPostTime publishedAt={post.attributes.published_at} url={'url'} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render the component with correct time', () => {
    const date = Date().toString();
    const post = buildPost({
      attributes: {
        published_at: date,
      },
    });

    const wrapper = render(<PopupPostTime publishedAt={post.attributes.published_at} url={'url'} />);
    const time = wrapper.queryByTitle(date);
    expect(time).toBeInTheDocument();
  });
});
