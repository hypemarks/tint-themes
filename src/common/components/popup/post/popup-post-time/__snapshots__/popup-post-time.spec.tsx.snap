// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PopupPostImage /> Component should render snapshot 1`] = `
<DocumentFragment>
  <a
    aria-label="published at null"
    class="popup-post-time"
    data-testid="link"
    href="url"
    rel="noopener noreferrer"
    style="color: rgb(228, 228, 228);"
    tabindex="0"
    target="_blank"
  />
</DocumentFragment>
`;
